src/Containers/DenseAxisArray.jl

```diff
@@ -148,17 +148,26 @@ function Base.get(
     return get(x.data, key[1] => key[2], default)
 end
 
-_abstract_vector(x::AbstractVector) = x
+function _abstract_vector(x::AbstractVector)
+    if eltype(x) <: CartesianIndex
+        error(
+            "Unsupported index type `CartesianIndex` in axis: $x. Cartesian " *
+            "indices are restricted for indexing into and iterating over " *
+            "multidimensional arrays.",
+        )
+    end
+    return x
+end
 
-_abstract_vector(x) = [a for a in x]
+_abstract_vector(x) = _abstract_vector(vec([a for a in x]))
```

Comment:
What is `vec` here for ?

Reply:
I don't remember. I've removed it to see which test broke.

Reply:
I added a test. It was for when the index set is an AbstractArray, not an AbstractVector.